### PR TITLE
feat: implement pattern matching to program

### PR DESCRIPTION
**Pull request by teran-joseluis**
## Description

This **PR** is on charge of implementing **Pattern Matching ** functionality to the main program that parses yamil code.

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: test/Parser/TestParserPatternMatching.hs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                                , PatternMatchLit (PLiteral (StringLiteral "One")) (IntLiteral 1) ]
+                                (OtherwiseLit (IntLiteral 2)))
+
+    it "parses multiple Bool -> String match cases separated by | and otherwise" $ do
+        parse parsePatternMatches "" "True -> \"Hello\" | otherwise -> \"Bye\"" `shouldBe` 
+                        Right (FullPatternMatch 
+                                [ PatternMatchLit (PLiteral (BoolLiteral True)) (StringLiteral "Hello") ]
+                                (OtherwiseLit (StringLiteral "Bye")))
+
+    -- it "parses multiple Double -> String match cases separated by | and otherwise" $ do
```
Comment (SebasthianSalpozz) [inline test/Parser/TestParserPatternMatching.hs:158]: You have a lot of code commented about the Double I think is a mistake Please fix it that

**Review by SebasthianSalpozz — CHANGES_REQUESTED** [review]
Add the end of the fix please do it a squash since most commits contain small changes

File: test/Parser/TestParserPatternMatching.hs (hunk 1 above)
Comment (teran-joseluis) [reply test/Parser/TestParserPatternMatching.hs]: Yeah, it was due to the origin branch was not merged yet into develop so the changes were old ones, but now it's okay.

**Comment by teran-joseluis** [comment]
> Add the end of the fix please do it a squash since most commits contain small changes

Okay! I'll do it after I gain your review.

**Comment by teran-joseluis** [comment]
> Add the end of the fix please do it a squash since most commits contain small changes

Okay! I'll do it after I gain your approve again.

**Review by SebasthianSalpozz — APPROVED** [review]
Good Job

**Review by fabianromeroclaros — APPROVED** [review]

**Pull request merged**